### DISCUSSION TITLE
Gatsby-like GraphQL API for SSG

### DISCUSSION THREAD

**Discussion by paulkre** [alephjs/aleph.js#351] [Ideas]
When it comes to developing static sites, Gatsby – with its powerful data querying system based on GraphQL – in my opinion still has a great advantage over Next.js. In SSG projects the data is usually defined in the file system of the repository. But this does not always mean that the structure of this data is simple. Often the data consists of complex relationships and dependencies which have to queryable by the application. Gatsby's GraphQL API makes working with these requirements very convenient. That can not be said about Next.js, unfortunately. 

Are there any plans to support something like Gatsby's data querying system for static site generation?

**Comment by mohsenkhanpour** [comment]
@ije 
I think if we create a separate package for GraphQL API would be super awesome!
It can be used with Aleph, however it can also be used with Next.js, Nuxt, Svelte, Vite, or whatever.
This API can be used to unify data fetching (both static/and server side) from markdown, json, yaml, wordpress, etc.

The API for such GraphQL API already exists in Node.js however it is tightly bound to the SSG/SSR frameworks like Gridsome or Gatsby.
https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-yaml
https://github.com/gridsome/gridsome/tree/master/packages/transformer-yaml

If we can port such API to Deno, we can make it available also for Next.js, Vite etc. in addition to Aleph.

Also it can be useful to provide many features that users want:
Image-optimization can be done through this API.
Markdown, and MDX sourcing.
Connecting to CMS.
